### PR TITLE
React and redux

### PR DESCRIPTION
**Pull request by arabiu033**
## changes Highlights :book: 

- Connect the app to the Redux store.
- Used useSelector to consume the state in required components
- Takes the inputs from the local state and generates a unique id and adds them to an object which is added to the store.
- Import required actions for dispatches.
- Fix linters and have the app running.

### CODE REVIEW

File: src/components/categories.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
 
-const Categories = () => (
-  <button type="button">Check Status</button>
-);
+const Categories = () => {
+  const dispatch = useDispatch();
+  return (
+    <button type="button" onClick={() => dispatch(status)}>Check Status</button>
+  );
+};
```
Comment (prolajumokeoni) [inline src/components/categories.js]: - [ ] Kindly make sure that the check status button works as expected,  Onclick it should dispatch `under construction`

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @arabiu033 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Git flow was correctly used 
✔️  Good PR title and Description



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/categories.js (hunk 1 above)
Comment (arabiu033) [reply src/components/categories.js]: Done :+1:

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @arabiu033 

Your project is complete! Great job for making the requested changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Validations correctly implemented
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**